addons/mail/static/src/discuss/call/common/call.js

```diff
@@ -109,17 +132,54 @@ export class Call extends Component {
     }
 
     get visibleMainCards() {
-        if (this.props.thread.activeRtcSession) {
+        const activeSession = this.props.thread.activeRtcSession;
+        if (activeSession) {
+            let videoType;
+            switch (activeSession.videoStreams.size) {
+                case 1:
+                    this.state.insetCard = false;
+                    if (
+                        activeSession.mainVideoStream !==
+                        activeSession.videoStreams.values().next().value
+                    ) {
+                        this.props.thread.activeRtcSession = false;
+                        activeSession.mainVideoStream = false;
+                    }
+                    break;
+                case 2:
+                    videoType =
+                        activeSession.videoStreams.get("screen") === activeSession.mainVideoStream
+                            ? "camera"
+                            : "screen";
+                    this.setInset(activeSession, videoType);
+                    break;
+            }
+
             return [
                 {
-                    key: "session_" + this.props.thread.activeRtcSession.id,
-                    session: this.props.thread.activeRtcSession,
+                    key: "session_" + activeSession.id,
+                    session: activeSession,
+                    videoStream: activeSession.mainVideoStream,
                 },
             ];
+        } else {
+            this.state.insetCard = false;
         }
         return this.visibleCards;
```

Comment:
Try to minimise nested code using [guard clauses](https://refactoring.guru/replace-nested-conditional-with-guard-clauses):
```js
if (!activeSession) {
    this.state.insetCard = null;
    return this.visibleCards;
}
// do the stuff that previously was in the if block
```